packages/connect/src/filter.ts

```diff
@@ -154,6 +157,15 @@ export class Filter {
       return false
     }
 
+    // Allow to dial localhost only if the port is different from all of those we're listening on
+    if (
+      isLocalhost(address.address, address.type) &&
+      this.announcedAddrs.some((announced) => announced.type === address.type && announced.port == address.port)
```

Comment:
```suggestion
      this.announcedAddrs.some((announced) => announced.type === address.type isLocalhost(announced.address, announced.type) && announced.port == address.port)
```

Reply:
Spotted that one as well, fixed already :)